File: 02_arrays/max_consecutive_ones/max_consecutive_ones.cpp
// Leetcode problem 485 - https://leetcode.com/problems/max-consecutive-ones/description/

 int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        int count = 0, max_count = 0;
        for(int i=0;i<n;i++){
            if(nums[i]==1){
                count++;
            }
            else{
                count = 0;
            }
            max_count = max(max_count, count);
        }
        return max_count;
    }
    

File: 02_arrays/max_consecutive_ones/max_consecutive_ones.java
public class max_consecutive_ones {

    static int findMaxConsecutiveOnes(int[] nums) {
        int count = 0;
        int maxCount = 0;

        for (int num : nums) {
            if (num == 1) {
                count++;
            } else {
                count = 0;
            }
            maxCount = Math.max(maxCount, count);
        }

        return maxCount;
    }

    // Test
    public static void main(String[] args) {
        int[] nums = {1, 1, 0, 1, 1, 1};
        System.out.println(findMaxConsecutiveOnes(nums));  // Output: 3
    }
}
